frontend/pages/[resourceType]/[id].tsx

```diff
@@ -3,13 +3,31 @@ import { Divider, Group, Stack, Tabs, Text } from '@mantine/core';
 import { GetServerSideProps, InferGetServerSidePropsType } from 'next';
 import { FhirArtifact } from '@/util/types/fhir';
 import BackButton from '../../components/BackButton';
+import { useEffect, useState } from 'react';
+import PrismRenderer from '../../prismLanguages/prismCQL';
 
 /**
  * Component which displays the JSON/ELM/CQL/narrative content of an individual resource using
  * Mantine tabs
  * @returns JSON/ELM/CQL/narrative content of the individual resource in a Prism component
  */
 export default function ResourceIDPage({ jsonData }: InferGetServerSidePropsType<typeof getServerSideProps>) {
+  const [decodedCql, setDecodedCql] = useState<string | null>(null);
+
+  // Overwrite Prism with our custom Prism that includes CQL as a language
+  useEffect(() => {
+    (window as any).Prism = PrismRenderer;
+  }, []);
+
+  useEffect(() => {
```

Comment:
In conjunction with my above comment and some of my recommendations regarding derived values in that [style guide](https://gist.github.com/mgramigna/bdc933fae2150216681c28d1f0f7aac7#hooks) that I made, I don't think `useEffect` is appropriate here since A) the content in the dependency array will never change and B) we can just derive the value in the comopnent itself.

If you want, you could memoize it instead so that, if other situations in this component cause a re-render, we don't re-compute the CQL:

```ts
const decodedCql = useMemo(() => {
  if (jsonData.resourceType === 'Measure') return null;
  const encodedCql = jsonData.content?.find(e => e.contentType === 'text/cql')?.data;
  return encodedCql ? Buffer.from(encodedCql, 'base64').toString() : null;
}, [jsonData]);
```

It's just an array lookup on something that will have like max 3 entries, so it's prob not that expensive. Although maybe on occasion the string could be a lot of bytes. idk

Reply:
Always forget about useMemo. Good tip!